google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb

```diff
@@ -202,17 +250,20 @@ def labels
         #
         def labels= new_labels
           raise ArgumentError, "Value must be a Hash" if new_labels.nil?
-          labels_map = Google::Protobuf::Map.new(:string, :string)
-          Hash(new_labels).each { |k, v| labels_map[String(k)] = String(v) }
-          update_grpc = @grpc.dup
```

Comment:
Nothing is getting lost here now that we are no longer `dup` ing the existing `@grpc` object?

Reply:
Nope. This refactor doesn't make any changes to behavior. The update is specifying a field mask. If anything, this new approach is a tiny bit more performant.

Reply:
OK, sounds good.